pennylane_lightning/core/bindings/Bindings_nb.hpp

```diff
@@ -636,13 +672,19 @@ OpsData<StateVectorT> createOpsList(
         &ops_matrices,
     const std::vector<std::vector<std::size_t>> &ops_controlled_wires,
     const std::vector<std::vector<bool>> &ops_controlled_values) {
-
     using ComplexT = typename StateVectorT::ComplexT;
-    using ParamT = typename StateVectorT::PrecisionT;
 
-    auto conv_matrices =
-        Pennylane::NanoBindings::Utils::convertMatrices<ComplexT, ParamT>(
-            ops_matrices);
+    // Convert ops_matrices to std::vector<std::vector<ComplexT>>
+    std::vector<std::vector<ComplexT>> conv_matrices(ops_matrices.size());
+
+    for (std::size_t i = 0; i < ops_matrices.size(); i++) {
+        if (ops_matrices[i].size() > 0) {
+            const auto *m_ptr =
+                PL_reinterpret_cast<const ComplexT>(ops_matrices[i].data());
+            const auto m_size = ops_matrices[i].size();
+            conv_matrices[i] = std::vector<ComplexT>(m_ptr, m_ptr + m_size);
+        }
+    }
```

Comment:
what's the motivation behind inlining `Pennylane::NanoBindings::Utils::convertMatrices` here?

Reply:
Could you also rewrite this with pre-defined `std::transform` from algorithm?